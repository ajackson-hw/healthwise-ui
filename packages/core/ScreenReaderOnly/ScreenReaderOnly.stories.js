import React from 'react'
import { storiesOf } from '@storybook/react'

import ScreenReaderOnly from './index'

storiesOf('core|Dev Utilities/Screen Reader Only', module).add(
  'with defaults',
  () => (
    <div>
      <p>Use a screen reader to hear the ScreenReaderOnly message.</p>
      <ScreenReaderOnly>This message is only visible for a screen reader.</ScreenReaderOnly>
    </div>
  ),
  {
    info: `Demonstrates basic rendering with defaults`,
  }
)
