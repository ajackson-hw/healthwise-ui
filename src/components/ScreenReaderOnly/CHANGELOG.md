# Change Log

## 1.1.3 - May 24, 2018

- Version bump. Testing publishing strategies.

## 1.1.2 - May 24, 2018

- Updated package-lock files
- Updated snapshots for newer version of Jest in `react-app-config`

## 1.1.1 - May 8, 2018

- Added `package-lock.json` files
- Update React peer dependency to `>=15.4.2` to remove warnings on React 16

## 1.1.0 - Initial release
