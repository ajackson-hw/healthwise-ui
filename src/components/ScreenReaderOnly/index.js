import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './ScreenReaderOnly.css'

class ScreenReaderOnly extends Component {
  render() {
    const { className, children, ...otherProps } = this.props
    const screenReaderClassName = classNames(
      'hw-screen-reader-only',
      styles.ScreenReaderOnly,
      className
    )

    return (
      <div className={screenReaderClassName} {...otherProps}>
        {children}
      </div>
    )
  }
}

ScreenReaderOnly.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
}

export default ScreenReaderOnly
