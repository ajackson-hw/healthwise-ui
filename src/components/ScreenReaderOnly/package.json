{
  "name": "@healthwise/screen-reader-only",
  "version": "1.1.3",
  "author": "Healthwise",
  "description": "Healthwise UI Assets - Renders hidden content to be read by screen readers",
  "license": "MIT",
  "homepage": "https://github.com/healthwise/healthwise-ui#readme",
  "main": "index.js",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "classnames": "^2.2.5",
    "prop-types": "^15.5.7"
  },
  "peerDependencies": {
    "react": ">=15.4.2"
  }
}
