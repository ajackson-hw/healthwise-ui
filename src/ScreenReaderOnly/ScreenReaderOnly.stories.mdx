import { Meta, Story, Preview } from '@storybook/addon-docs/blocks';

import ScreenReaderOnly from './index';

<Meta title="core|Dev Utilities/Screen Reader Only" component={ScreenReaderOnly} />

# ScreenReaderOnly

Demonstrates basic rendering with defaults.

<Preview>
  <Story name="default">
    <div>
      <p>Use a screen reader to hear the ScreenReaderOnly message.</p>
      <ScreenReaderOnly>This message is only visible for a screen reader.</ScreenReaderOnly>
    </div>
  </Story>
</Preview>
