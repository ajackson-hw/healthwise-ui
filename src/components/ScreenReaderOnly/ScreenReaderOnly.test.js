/* global describe it expect */
import React from 'react'
import renderer from 'react-test-renderer'

import ScreenReaderOnly from './index'

describe('<ScreenReaderOnly />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ScreenReaderOnly>Test screen reader message</ScreenReaderOnly>)
    expect(tree).toMatchSnapshot()
  })

  it('can render HTML content', () => {
    const tree = renderer.create(
      <ScreenReaderOnly>
        <div>Test screen reader message</div>
      </ScreenReaderOnly>
    )
    expect(tree).toMatchSnapshot()
  })
})
