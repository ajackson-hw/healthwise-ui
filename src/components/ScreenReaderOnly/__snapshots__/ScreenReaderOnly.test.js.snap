// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenReaderOnly /> can render HTML content 1`] = `
<div
  className="hw-screen-reader-only ScreenReaderOnly"
>
  <div>
    Test screen reader message
  </div>
</div>
`;

exports[`<ScreenReaderOnly /> renders correctly 1`] = `
<div
  className="hw-screen-reader-only ScreenReaderOnly"
>
  Test screen reader message
</div>
`;
